components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/token/APIMJWTGenerator.java

```diff
@@ -161,9 +160,62 @@ public String buildBody(JwtTokenInfoDTO jwtTokenInfoDTO) throws APIManagementExc
         claims.put("tierInfo", jwtTokenInfoDTO.getSubscriptionPolicyDTOList());
         claims.put("application", jwtTokenInfoDTO.getApplication());
         claims.put("keytype", jwtTokenInfoDTO.getKeyType());
-        claims.put("consumerKey" , jwtTokenInfoDTO.getConsumerKey());
+        claims.put("consumerKey", jwtTokenInfoDTO.getConsumerKey());
         claims.put("backendJwt", jwtTokenInfoDTO.getBackendJwt());
 
         return claims;
     }
+
+    public byte[] signJWT(String assertion) throws APIManagementException {
+
+        PrivateKey privateKey;
+        KeyStoreManager superTenantKSM = KeyStoreManager.getInstance(MultitenantConstants.SUPER_TENANT_ID);
+        try {
+            privateKey = superTenantKSM.getDefaultPrivateKey();
+        } catch (Exception e) {
+            String msg = "Error while obtaining private key for super tenant";
+            log.error("Error while obtaining private key for super tenant", e);
+            throw new APIManagementException(msg, e);
+        }
+        return APIUtil.signJwt(assertion, privateKey, signatureAlgorithm);
+    }
+
+    public String buildHeader() throws APIManagementException {
+
+        String jwtHeader = null;
+
+        //if signature algo==NONE, header without cert
+        if (NONE.equals(signatureAlgorithm)) {
+
+            StringBuilder jwtHeaderBuilder = new StringBuilder();
+            jwtHeaderBuilder.append("{\"typ\":\"JWT\",");
+            jwtHeaderBuilder.append("\"alg\":\"");
+            jwtHeaderBuilder.append(APIUtil.getJWSCompliantAlgorithmCode(NONE));
+            jwtHeaderBuilder.append('\"');
+            jwtHeaderBuilder.append('}');
+
+            jwtHeader = jwtHeaderBuilder.toString();
+
+        } else if (SHA256_WITH_RSA.equals(signatureAlgorithm)) {
+            jwtHeader = addCertToHeader();
+        }
+        return jwtHeader;
+    }
+
+    /**
+     * Helper method to add public certificate to JWT_HEADER to signature verification.
```

Comment:
Why we adding public certificate to the header? Correctly highligh what we adding to the header.